engine/src/main/java/org/apache/apex/engine/plugin/ApexPluginManager.java

```diff
@@ -187,5 +188,19 @@ PluginInfo getPluginInfo(DAGExecutionPlugin plugin)
     {
       ApexPluginManager.this.register(type, handler, owner);
     }
+
+    @Override
+    public DAG getDAG()
+    {
+      return clonedDAG;
+    }
+  }
+
+  @Override
+  public <T> void dispatch(RegistrationType<T> registrationType, T data)
```

Comment:
Suggest breaking this into two methods, dispatch can call another abstract method like dispatchEvent that the implementation implements. The implementation will not have to call super.